acme/acme/messages.py

```diff
@@ -756,7 +711,98 @@ class OrderResource(ResourceWithURI):
                                                        omitempty=True)
 
 
-@Directory.register
-class NewOrder(Order):
-    """New order."""
-    resource_type = 'new-order'
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", "acme.messages.Directory.register", DeprecationWarning)
+    warnings.filterwarnings("ignore", "acme.fields.resource", DeprecationWarning)
+
+    @Directory.register
```

Comment:
Reorganizing the declarations like this is supremely fugly but I couldn't find a nice way to apply `filterwarnings` to the entire module without it leaking out.

One possibility I considered was to do something like:

```python
warnings_capture = warnings.catch_warnings()
warnings_capture.__enter__()
warnings.filterwarnings("ignore", "acme.messages.Directory.register", DeprecationWarning)
warnings.filterwarnings("ignore", "acme.fields.resource", DeprecationWarning)
# ...
# rest of the module code goes here
# ...
warnings_capture.__exit__()
```

Let me know what you think.

Reply:
I think you took the best approach. Sure, there's a larger diff now, but I think the resulting code is simpler and more conventional which I personally think is more important.